code/game/objects/items/devices/megaphone.dm

```diff
@@ -27,7 +27,7 @@
 		return
 
 	var/message = copytext(sanitize(input(user, "Shout a message?", "Megaphone", null)  as text),1,MAX_MESSAGE_LEN)
-	if(!message)
+	if(!message || message == "")
```

Comment:
So a proc like this 
    
    /proc/isemptystring(string)
        if(!replacetext(string, " ", "")) 
            return 1
        return 0

would work to check for an empty message?